### ISSUE TITLE
Stop Methods Not Being Called on Exit

### ISSUE THREAD

**Issue by primesoftware** [eclipse/vert.x#1603]
I've made a simple test to figure out why none of my stop methods are getting called on exit (Ctrl-C from command line).

`public class SimpleServer extends AbstractVerticle {
    private HttpServer httpServer = null;

    @Override
    public void stop()
    {
        System.out.println("tallyho!!");
    }

    @Override
    public void start() throws Exception
    {
        Router mainRouter = Router.router(vertx);
        mainRouter.delete("/something").handler(routingCtx -> {});
        httpServer = vertx.createHttpServer().requestHandler(mainRouter::accept);
        httpServer.listen(9999);
        System.out.println("started!!");
    }

    public static void main(String[] args) throws InterruptedException
    {
        Vertx vertx = Vertx.vertx();
        vertx.deployVerticle("com.mykpg.SimpleServer", stringAsyncResult ->
        {
            if (stringAsyncResult.succeeded())
            {
                //vertx.close() works on my simple example but not my 3 verticle (1 worker) actual app
                System.out.println(stringAsyncResult.result());
            }
            else
            {
                System.out.println(stringAsyncResult.cause().getMessage());
            }
        });
    }
}`

I don't get any error messages even though I've Log4J2 set to debug. I've disabled Log4J2 and same behavior. vertx.close() only works in this very simple example. On my 3 verticle (1 worker) actual app vertx.close() does nothing. Any ideas?

**Comment by vietj** [comment]
shutdown hooks are added by the Vert.x CLI commands

**Comment by vietj** [comment]
this seem to run with a main ?

**Comment by primesoftware** [comment]
@vietj I was told by Tim on the Goggle Group that Ctrl-C should terminate the vertx app and call the stop methods since there is a built-in hook already setup to do that. For whatever reason however, Ctrl-C kills my app without calling stop methods be they blocking or not.

I've worked around this by setting a periodic timer to check for the existence of a PID file every second. If it gets deleted, vertx.close() is called everything works. I've read in some places that sometimes on Linux Ctrl-C actions are an abrupt halt that do not get passed to the hooks. I am not sure if that true for all, but in my case it certainly is. 

Yes, this runs with a main b/c I'm using the fat jar method, not the Vertx command. I'm going to close this since I've got a work around.....
